### PR TITLE
Acid-related nerfs

### PR DESCRIPTION
**Pull request by Coolthulhu**
Changed the way acid fields work:
First off, this is mostly a nerf for acid vs. player damage. You don't need env. protection not to have legs melted off, but if you don't have env. protection, you'll need to dance around. Acid fields now add "corroding" effect which causes minor damage and reduces effective acid resistance of armor.
However if you do manage to lose legs, the acid will now burn the rest of the body.

Slightly nerfed acid effect vs. monsters so that it doesn't melt them. "Realistically" zombies should be rather resistant to it.

Changed corrosive zombie's splashback a bit. Now it produces low-density fields. Using a cutting weapon will make it splatter around more. Using unarmed one will burn hands.
Will print a message if any of the fields is placed on the player.

Nerfed kevlar's acid resistance a bit. 7->5

Nerfed survivor armors' environmental protection (from 10, except for masks and goggles):
- Regular and light armors get 3
- Heavy armors get 5
- Winter gear gets 5
- Masks and goggles keep 15
- Firegear and water gear keeps 10 (and now gets a reason to be used)

### CODE REVIEW

**Comment by Zireael07** [comment]
Thanks for that firegear and watergear part!!!

File: src/mondefense.cpp [hunk 1]
```diff
@@ -47,41 +47,75 @@ void mdefense::zapback( monster *const m, Creature *const source, projectile con
...
+        if( foe->weapon.is_cutting_weapon() ) {
+            num_drops += rng( 1, 2 );
+        }
+
+        if( foe->unarmed_attack() ) {
+            damage_instance const burn {
+                DT_ACID, static_cast<float>( rng( 1, 5 ) )
+            };
+
+            if( one_in( 2 ) ) {
```
Comment (kevingranade) [inline src/mondefense.cpp]: Future note, unarmed attacks should pick the attacking limb so that styles can adjust this.

**Comment by BevapDin** [comment]
The mods (Arcana, Medival, probably more) don't have the new (mandatory) material property, @chaosvolt to the rescue!

**Comment by chaosvolt** [comment]
Doh. Oh shit. Guess I'll PR that then. ;w;

**Pull request merged**